src/api/api_missions.nit

```diff
@@ -35,6 +36,40 @@ end
 
 class APIMission
 	super MissionHandler
+	super AuthHandler
+
+	redef fun post(req, res) do
+		var player = get_player(req, res)
+		if player == null then return
+		var mission = get_mission(req, res)
+		if mission == null then return
+		var post = req.post_args
+		if not post.has_key("source") then
+			res.error 400
+			print "No code provided for submission"
+			return
+		end
+		var code = post["source"]
+		if not post.has_key("lang") then
+			res.error 400
+			print "No lang provided for submission"
+			return
+		end
+		var engine = post["lang"]
+		if not config.engine_map.has_key(engine) then
```

Comment:
Yup, this should be checked for the mission.

But I think I understand what you tried to do here.

One solution could be to list the `accepted_languages` in the mission then match it with the query language. So one can decide that a specific mission can be answered in multiple languages.

In a first time I suggest to juste get the language from the `Mission::accepted_language` (just one).